tensorboard/components/tf_trace_viewer/tf-trace-viewer.html

```diff
@@ -53,6 +53,48 @@
 <script>
   "use strict";
 
+  /* streaming mode helper functions */
+  var ZOOM_RATIO = 8;      // Amount of zooming allowed before re-fetching.
```

Comment:
One issue with declaring variables and functions here is that they are placed in the global scope of TensorBoard. Could you please place these functions and constants within a TypeScript file within this directory? Say `tf-trace-viewer-helper.ts`. And then surround these properties with a namespace. It should look something like this:

```typescript
/**
 * @fileoverview Helper utilities for the trace viewer within TensorBoard's profile plugin.
 */

module tf.profile.traceViewer {
  /** Amount of zooming allowed before re-fetching. */
  export const ZOOM_RATIO = 8;

  /** Minimum safety buffer relative to viewport size. */
  export const PRESERVE_RATIO = 2;

  /** Amount to fetch relative to viewport size. */
  export const FETCH_RATIO = 3;

  export interface Range {
    min: number;
    max: number;
  }

  /**
   * A brief docstring ...
   */
  export function expand(range: Range, scale: number): Range {
    var width = range.max - range.min;
    var mid = range.min + width / 2;
    return {
      min: mid - scale * width / 2,
      max: mid + scale * width / 2,
    };
  }
  /**
   * A brief docstring ...
   */
  export function within(range: Range, bounds: Range): boolean {
    return bounds.min <= range.min && range.max <= bounds.max;
  }
  /**
   * A brief docstring ...
   */
  export function length(range: Range): number {
    return range.max - range.min;
  }
  /**
   * A brief docstring ...
   */
  function intersect(range: Range, bounds: Range): Range {
    return {
      min: Math.max(range.min, bounds.min),
      max: Math.min(range.max, bounds.max),
    };
  }
  /**
   *  Compute the {min, max} range of a trackView.
   *  TODO: Spec out an interface for the trackView.
   */
  function trackViewRange(trackView): Range {
    var xfm = trackView.viewport.currentDisplayTransform;
    const pixelRatio = window.devicePixelRatio || 1;
    const devicePixelWidth = pixelRatio * trackView.viewWidth_;
    return {
      min: xfm.xViewToWorld(0),
      max: xfm.xViewToWorld(devicePixelWidth),
    };
  }
  
} // close module tf-profile
```

And then, you'd include the script within the HTML file. See how `utils.js` is used here:
https://github.com/tensorflow/tensorboard/tree/master/tensorboard/plugins/profile/tf_op_profile

Within the HTML, you'd access properties via the namespace, ie `tf.profile.traceViewer.trackViewRange`

Reply:
PHAL